recipes/fmilib/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+
+
+class PackageConan(ConanFile):
+    name = "fmilib"
+    description = "C library for importing FMUs"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/modelon-community/fmi-library"
+    topics = ("fmi", "fmi-standard", "fmu")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_fmus": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_fmus": False
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("fmi1/1.0.1")
+        self.requires("fmi2/2.0.4")
+        if self.version >= Version("3.0a2"):
+            self.requires("fmi3/3.0.1")
+        self.requires("expat/2.5.0")
+        self.requires("minizip/1.2.13")
+        self.requires("zlib/[>=1.2.11 <2]")
+        # c99_snprintf -> should be externalised
+
+    def validate(self):
+        # https://github.com/modelon-community/fmi-library/issues/93
+        if self.settings.arch not in ["x86", "x86_64"] and Version(self.version).major < 3:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support architecture "
+                f"'{self.settings.arch}' on {self.settings.os}")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        tc.variables["FMILIB_BUILD_STATIC_LIB"] = not self.options.shared
+        tc.variables["FMILIB_BUILD_SHARED_LIB"] = self.options.shared
+        tc.variables["FMILIB_BUILD_TESTS"] = self.options.with_fmus
+        tc.variables["FMILIB_FMI_STANDARD_HEADERS"] = posixpath.join(self.build_folder, "fmis").replace("\\", "/")
+        tc.variables["FMILIB_GENERATE_DOXYGEN_DOC"] = False
+
+        # The variable is an option only if the following condition is true
+        if not self.options.shared and not self.settings.os in ["Windows", "Macos"]:
+            tc.variables["FMILIB_BUILD_FOR_SHARED_LIBS"] = self.options.get_safe("fPIC", False)
+
+        if is_msvc(self):
+            tc.variables["FMILIB_BUILD_WITH_STATIC_RTLIB"] = is_msvc_static_runtime(self)
+
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0026"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0045"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0046"] = "OLD"
```

Comment:
```suggestion
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0046"] = "OLD"
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```
Needed because the project's CMake policy version is less than 3.12.

Reply:
I added this to the newest commit.